submitting-changes.chapter.md: explain that purple arrows are manual

a-m-joseph opened the pull request:
The documentation for this diagram explains that the blue arrows are
automatic processes which happen every six hours.  There is no
explanation about how the purple arrows happen or how often.

As a new contributor to nixpkgs, I incorrectly assumed that the purple
arrows were also automatic processes (they aren't), which left me sort
of confused about what the whole scheme was accomplishing.

Recently I went through the github history to see how often these
events happen, and realized that the purple arrows are (a) triggered
manually by a nixpkgs project member and (b) happen much, much, much
less frequently than every six hours.

Now everything makes a lot more sense.  I suggest the wording change
in this commit, or something similar, to save future contributors the
same confusion that I experienced.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
 
~~I deleted the other checkbox items because this is a documentation-only change.  `cd doc; nix-shell` wants to rebuild 317 packages so I will let it run overnight and tick the box in the morning to confirm that I didn't make any markdown syntax errors.~~ rebuild finished.

Commit:
submitting-changes.chapter.md: explain that purple arrows are manual

The documentation for this diagram explains that the blue arrows are
automatic processes which happen every six hours.  There is no
explanation about how the purple arrows happen or how often.

As a new contributor to nixpkgs, I incorrectly assumed that the purple
arrows were also automatic processes (they aren't), which left me sort
of confused about what the whole scheme was accomplishing.

Recently I went through the github history to see how often these
events happen, and realized that the purple arrows are (a) triggered
manually by a nixpkgs project member and (b) happen much, much, much
less frequently than every six hours.

Now everything makes a lot more sense.  I suggest the wording change
in this commit, or something similar, to save future contributors the
same confusion that I experienced.

mweinelt commented:
The description is technically correct. I will leave the specific wording to others.

vcunat commented:
I'm unsure about the last sentence.  For practical searching it might be better to suggest something like https://github.com/NixOS/nixpkgs/pulls?q=head%3Astaging-next even though it's not perfect either.  Merges by me usually don't contain "from NixOS/staging-next" :-)  though in this case it might make more sense than in others.

Commit:
submitting-changes.chapter.md: avoid being specific

There is some doubt as to exactly how to enumerate all the merges from
one branch to another reliably.  In the meantime, let's be a little
more vague.

a-m-joseph commented:
@vcunat raises an important point, and I don't know exactly how to reliably enumerate these data.  So, instead, I have pushed a commit which dodges the issue instead of solving it :)

vcunat commented:
No, https://github.com/orgs/NixOS/teams/nixpkgs-committers can push any fast-forward changes to nixpkgs master, though avoiding PRs [is discouraged](https://github.com/NixOS/nixpkgs/issues/118661).  Pull requests have been created for staging* iterations for 3&ndash;4 years apparently.  Anyway, it looks like I'm digressing significantly 😄 

Practical period has recently (like a year) been from a few days, up to a few weeks, I believe.

a-m-joseph commented:
Ping

vcunat commented:
Right, this got pushed down in my todo list.  Let's take it as it is.

a-m-joseph commented:
> Right, this got pushed down in my todo list.

No worries, my own todo list has exploded lately...
